File: sdkwork-backend-sdk-typescript/src/api/visit-history.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultBoolean, PlusApiResultListPlusVisitHistoryVO, PlusApiResultPagePlusVisitHistoryVO, PlusApiResultPlusVisitHistoryVO, PlusVisitHistoryForm, QueryListForm } from '../types';


export class VisitHistoryApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Update Visit History */
  async update(body: PlusVisitHistoryForm): Promise<PlusApiResultPlusVisitHistoryVO> {
    return this.client.put<PlusApiResultPlusVisitHistoryVO>(backendApiPath(`/visit_history`), body);
  }

/** Create Visit History */
  async create(body: PlusVisitHistoryForm): Promise<PlusApiResultPlusVisitHistoryVO> {
    return this.client.post<PlusApiResultPlusVisitHistoryVO>(backendApiPath(`/visit_history`), body);
  }

/** List Visit Histories by Page */
  async listByPage(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusVisitHistoryVO> {
    return this.client.post<PlusApiResultPagePlusVisitHistoryVO>(backendApiPath(`/visit_history/list`), body, params);
  }

/** List All Visit Histories */
  async listAllEntities(body?: QueryListForm): Promise<PlusApiResultListPlusVisitHistoryVO> {
    return this.client.post<PlusApiResultListPlusVisitHistoryVO>(backendApiPath(`/visit_history/list/all`), body);
  }

/** Get Visit History by ID */
  async getById(id: string | number): Promise<PlusApiResultPlusVisitHistoryVO> {
    return this.client.get<PlusApiResultPlusVisitHistoryVO>(backendApiPath(`/visit_history/${id}`));
  }

/** Delete Visit History */
  async delete(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/visit_history/${id}`));
  }
}

export function createVisitHistoryApi(client: HttpClient): VisitHistoryApi {
  return new VisitHistoryApi(client);
}

File: sdkwork-backend-sdk-go/api/visit_history.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type VisitHistoryApi struct {
    client *sdkhttp.Client
}

func NewVisitHistoryApi(client *sdkhttp.Client) *VisitHistoryApi {
    return &VisitHistoryApi{client: client}
}

// Update Visit History
func (a *VisitHistoryApi) Update(body sdktypes.PlusVisitHistoryForm) (sdktypes.PlusApiResultPlusVisitHistoryVO, error) {
    raw, err := a.client.Put(BackendApiPath("/visit_history"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusVisitHistoryVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusVisitHistoryVO](raw)
}

// Create Visit History
func (a *VisitHistoryApi) Create(body sdktypes.PlusVisitHistoryForm) (sdktypes.PlusApiResultPlusVisitHistoryVO, error) {
    raw, err := a.client.Post(BackendApiPath("/visit_history"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusVisitHistoryVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusVisitHistoryVO](raw)
}

// List Visit Histories by Page
func (a *VisitHistoryApi) ListByPage(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusVisitHistoryVO, error) {
    raw, err := a.client.Post(BackendApiPath("/visit_history/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusVisitHistoryVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusVisitHistoryVO](raw)
}

// List All Visit Histories
func (a *VisitHistoryApi) ListAllEntities(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusVisitHistoryVO, error) {
    raw, err := a.client.Post(BackendApiPath("/visit_history/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusVisitHistoryVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusVisitHistoryVO](raw)
}

// Get Visit History by ID
func (a *VisitHistoryApi) GetById(id string) (sdktypes.PlusApiResultPlusVisitHistoryVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/visit_history/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusVisitHistoryVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusVisitHistoryVO](raw)
}

// Delete Visit History
func (a *VisitHistoryApi) Delete(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/visit_history/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}
